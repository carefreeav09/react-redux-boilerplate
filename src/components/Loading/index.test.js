import React from 'react';
import Enzyme, {shallow} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';

import Loader from './Loader';
import Loading from './index';

Enzyme.configure({adapter: new Adapter()});

let wrapper;

describe('<Loading />', () => {

  it("Should export Layout", () => {
    const props = {
      error: {},
      pastDelay: false
    };
    wrapper = shallow(<Loading {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should check is past Delay is true", () => {
    const props = {
      pastDelay: true
    };
    wrapper = shallow(<Loading {...props} />);
    expect(wrapper.contains(<Loader/>)).toBe(true);
  })
});
