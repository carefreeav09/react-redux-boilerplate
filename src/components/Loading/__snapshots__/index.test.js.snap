// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> Should export Layout 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading
    error={Object {}}
    pastDelay={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Error! Please refresh the page",
      "className": "loader-container",
    },
    "ref": null,
    "rendered": "Error! Please refresh the page",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Error! Please refresh the page",
        "className": "loader-container",
      },
      "ref": null,
      "rendered": "Error! Please refresh the page",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
